src/MostUsed/CLI.hs

```diff
@@ -27,13 +28,28 @@ withInfo opts h d f =
     info (helper <*> opts) $ header h <> progDesc d <> footer f
 
 parseOptions :: Parser Options
-parseOptions = Options <$> parseIncludeFirstArgument <*> parseDebug
+parseOptions = Options
+    <$> parseIncludeFirstArgument
+    <*> parseDebug
+    <*> parseShell
 
 parseIncludeFirstArgument :: Parser [String]
 parseIncludeFirstArgument = many $ strOption $
     long "include-first-argument"
+    <> metavar "command_name"
     <> help "Count this command with its first argument (can be specified more than once)"
 
+parseShell :: Parser Shell
+parseShell = fmap shell $ strOption $
+    long "shell"
+    <> metavar "[bash | zsh]"
+    <> help "Which type of shell history to parse"
+
+shell :: String -> Shell
+shell "bash" = Bash
+shell "zsh" = Zsh
+shell _ = error "--shell can only take 'bash' or 'zsh'"
```

Comment:
I wonder if this should be a complete function, and this should be like `Unsupported String` or something? Although that would mean you'd need to support `Unsupported` as a shell type elsewhere so maybe this really is the best place for this error to be thrown.